### PR TITLE
Change ATF test reports folder structure

### PR DESCRIPTION
**Pull request by dboltovskyi**
Main output of ATF is test reports. ATF provide such reports:

- Console logs - name of test cases with execution status FAILED/PASSED.
- SDL logs - SDL log received by ATF via telnet logger.
- Transport logs - all received and sent data from/to SDL. Transport log can have default view - messages on protocol level (except binaries) and full view - log will be expanded with packages: streamings, service messages, heartbeat, json files.
- Detailed report - all expected and sent massages related to test cases with expected and actual result.Also contains info about test cases: name, time, sequence, status, duration.

This proposal is about creating clear and useful structure of test scripts reports.

### CODE REVIEW

File: proposals/NNNN-Change_ATF_test_reports_folder_structure.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Change ATF test reports folder structure
+
+* Proposal: [SDL-NNNN](NNNN-Change_ATF_test_reports_folder_structure.md)
+* Author: [Irina Getmanets](https://github.com/GetmanetsIrina)
+* Status: **Awaiting review**
+* Impacted Platforms: ATF
+
+## Introduction
+
+For now ATF reports folder has inconvenient nested structure. ATF creates structure of folders from relative path to script, e.g. runned script fom test_scripts/API/Navidation/Subscriptions/.
```
Comment (LuxoftAKutsan) [inline proposals/NNNN-Change_ATF_test_reports_folder_structure.md]: @dboltovskyi @GetmanetsIrina It is too big and to detailed for introduction. 
Introduction must contain hight level description of changes.
Like : 
Main output of ATF is test reports. 
ATF provide such reports : 
  - Console logs
  - SDL logs
  - Transport  logs
  - Detailed report

This proposal is about creating clear and useful structure of test scripts reports. 
Multiple running of test script should not override old reports. 
Structure of reports should be clear for easy searching certain report.

File: proposals/NNNN-Change_ATF_test_reports_folder_structure.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+With current implementation will be created the next structure
+<pre><code>
```
Comment (LuxoftAKutsan) [inline proposals/NNNN-Change_ATF_test_reports_folder_structure.md]: @dboltovskyi @GetmanetsIrina There should not be any xml tags. Use Markdown formating

File: proposals/NNNN-Change_ATF_test_reports_folder_structure.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  ->Reports_timestamp
+    ->test_scripts
+      ->API
+      	->Navigation
+	  ->Subscriptions
+</pre></code>
+
+## Motivation
+
+Create more convenient reports structure, less nested.
```
Comment (LuxoftAKutsan) [inline proposals/NNNN-Change_ATF_test_reports_folder_structure.md]: @dboltovskyi @GetmanetsIrina please expand motivation ( you can move information from introduction)

File: proposals/NNNN-Change_ATF_test_reports_folder_structure.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+## Proposed solution
+
+The solution is to create new structure:
+<pre><code>TestingReports
+	-> ScriptName_timestamp
```
Comment (LuxoftAKutsan) [inline proposals/NNNN-Change_ATF_test_reports_folder_structure.md]: @dboltovskyi @GetmanetsIrina please specify format of timestamp

File: proposals/NNNN-Change_ATF_test_reports_folder_structure.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+n/a
+
+## Impact on existing code
+
+Impact on ATF reporting functionality.
+
+## Alternatives considered
+
+n/a
```
Comment (LuxoftAKutsan) [inline proposals/NNNN-Change_ATF_test_reports_folder_structure.md]: @dboltovskyi @GetmanetsIrina Alternative - leave as is

File: proposals/NNNN-Change_ATF_test_reports_folder_structure.md (hunk 1 above)
Comment (GetmanetsIrina) [reply proposals/NNNN-Change_ATF_test_reports_folder_structure.md]: @LuxoftAKutsan , please find changes in commit 14e20e522aa7592aebf356eec293d24797f7802e.

File: proposals/NNNN-Change_ATF_test_reports_folder_structure.md (hunk 2 above)
Comment (GetmanetsIrina) [reply proposals/NNNN-Change_ATF_test_reports_folder_structure.md]: @LuxoftAKutsan , please find changes in commit 14e20e522aa7592aebf356eec293d24797f7802e.

File: proposals/NNNN-Change_ATF_test_reports_folder_structure.md (hunk 3 above)
Comment (GetmanetsIrina) [reply proposals/NNNN-Change_ATF_test_reports_folder_structure.md]: @LuxoftAKutsan , please find changes in commit 14e20e522aa7592aebf356eec293d24797f7802e.

File: proposals/NNNN-Change_ATF_test_reports_folder_structure.md (hunk 4 above)
Comment (GetmanetsIrina) [reply proposals/NNNN-Change_ATF_test_reports_folder_structure.md]: @LuxoftAKutsan , please find changes in commit 14e20e522aa7592aebf356eec293d24797f7802e.

File: proposals/NNNN-Change_ATF_test_reports_folder_structure.md (hunk 5 above)
Comment (GetmanetsIrina) [reply proposals/NNNN-Change_ATF_test_reports_folder_structure.md]: @LuxoftAKutsan , please find changes in commit 14e20e522aa7592aebf356eec293d24797f7802e.

**Comment by LuxoftAKutsan** [comment]
@GetmanetsIrina very good. 
Also please add short description of each report and also split transport logs to 2 parts : 
 - packets 
 - messages only

**Comment by GetmanetsIrina** [comment]
@LuxoftAKutsan, please find the update in commit 418ed2b404a10233f52826eb720052896cf44518

**Comment by AStasiuk** [comment]
@theresalech proposal is ready for review

File: proposals/NNNN-Change_ATF_test_reports_folder_structure.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+* Console logs - name of test cases with execution status FAILED/PASSED.
+* SDL logs - SDL log received by ATF via telnet logger.
+* Transport logs -  all received and sent data from/to SDL. Transport log can have default view - messages on protocol level (except binaries) and full view - log will be expanded with packages: streamings, service messages, heartbeat, json files.   
+* Detailed report - all expected and sent massages related to test cases with expected and actual result.Also contains info about test cases: name, time, sequence, status, duration.
+
+This proposal is about creating clear and useful structure of test scripts reports.
+
+## Motivation
+
+Create more convenient reports structure, less nested. Multiple running of test script should not override old reports. Structure of reports should be clear for easy searching certain report.
```
Comment (theresalech) [inline proposals/NNNN-Change_ATF_test_reports_folder_structure.md]: Please update second sentence to "Running multiple test scripts should not override old reports."

Please update third sentence to "Structure of reports should be clear, so you can easily search for a certain report."

Comment (GetmanetsIrina) [reply proposals/NNNN-Change_ATF_test_reports_folder_structure.md]: @theresalech, updated, please find changes in commit 397541b6fb295b86ba1b3bf317b0a5035ff009cd.

**Comment by GetmanetsIrina** [comment]
@theresalech, thanks, fixed according to comments in commit 397541b6fb295b86ba1b3bf317b0a5035ff009cd.

**Pull request merged**